scripts/cpplint.py

```diff
@@ -224,8 +224,9 @@
     'readability/inheritance',
     'readability/multiline_comment',
     'readability/multiline_string',
-    'readability/namespace',
+    'readability/identifier_spacing',
```

Comment:
I don't think we should/need to skip these?

Reply:
Identifier-spacing was added for the rule about `:` when inheriting. Previously, this rule was labeled `readability/naming`, but there are other useful rules with this label that we want to keep enabled.

There is no need to keep the `:` rule enabled, as it is now enforced by clang-format. There's no need to check it again here.

Reply:
See https://github.com/diffblue/cbmc/pull/1324#discussion_r144848784 -- I'd be fine with you disabling rules where `cpplint` now wrongly reports a mistake just because `clang-format` does it slightly differently. But I'd rather not have rules disabled just because we can auto-generate the correct code as I fail to see the value of doing so.

Reply:
If we're saying "whatever clang-format does is correct" then I don't see the point in doing extra duplicated work in this script. The two outcomes are:
- the script gives the same results as clang-format (we did unnecessary work), or
- the script flags up a false positive and we defer to clang-format anyway (we also did unnecessary work)
Besides, it's better separation of concerns to let clang-format do 'just style stuff' and to restrict this script to more semantic things.

Reply:
You're highlighting how much each of us trust `clang-format`: you seem to fully trust it, whereas I'd rather not trust it just yet. Hence I'd happily add a third item "clang-format got it wrong and the script flags this."

I'll downgrade my "request for changes" to a comment and let someone else be the arbiter.

Reply:
My feeling is that we should only disable the rules that clang-format *cannot* meet in any configuration.  Not everyone will use clang-format and we will still be using the linter on CI.  Hell; maybe we should add something to the end of the linting output so that if it finds errors it says, "why not use scripts/whatever.sh to fix this".